src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/UsersController.kt

```diff
@@ -163,5 +163,7 @@ class UsersController(
       UserQualification.lao -> uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.LAO
       UserQualification.emergency -> uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.EMERGENCY
       UserQualification.esap -> uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.ESAP
+      UserQualification.mentalHealthSpecialist -> uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.MENTAL_HEALTH_SPECIALIST
```

Comment:
There seem to be a lot of converting back and forth of enums. I recently made a tweak to `PlacementApplicationDecision` to keep the conversion logic in the enum itself, which @davidatkinsuk made some suggestions to tweak (which I forgot to implement). We could do something similar for `UserQualification` like so:

```kotlin
enum class UserQualification(val apiValue: uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification) {
  WOMENS(uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification.womens),
  PIPE(uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification.pipe),
  LAO(uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification.lao),
  ESAP(uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification.esap),
  EMERGENCY(uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification.emergency);

  companion object {
    fun valueOf(apiValue: uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification): UserQualification? {
      for (e in entries) {
        if (e.apiValue == apiValue) {
          return e
        }
      }
      return null
    }
  }
}
```

That way we could do (for example):

```kotlin
UserQualification.mentalHealthSpecialist.apiValue //=> UserQualification.mentalHealthSpecialist -> uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.MENTAL_HEALTH_SPECIALIST
```

And:

```kotlin
UserQualification.valueOf(uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification.MENTAL_HEALTH_SPECIALIST) // => UserQualification.mentalHealthSpecialist
```

Reply:
This looks very neat, but I think I'm missing part of the puzzle, as the enum is defined in `package uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.UserQualification` which is auto-generated from the OpenApi spec at https://github.com/ministryofjustice/approved-premises-api/blob/e343d80d274904f6dc6bdb9f6e50ba8fc6421eed/src/main/resources/static/_shared.yml#L2956-L2965
So does your example `enum class UserQualification` over-ride that built definition? If so how?

I'm planning to merge this work as-is -- and then circle back "later" to make this improvement...

Reply:
Nah, so we have two `UserQualification` enums - an API one, and a JPA one, we spend a lot of time translating between the two, so it makes sense to have one source of truth for both. I did a similar piece of work here https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1733